### PR TITLE
Basic setup changes

### PR DESCRIPTION
**Pull request by maks-io**
I added prettier + a config file for prettier to the repo. There will be more pull requests coming from my side in the upcoming days, and i think it makes sense that every contributor of this project uses the same formatting settings.
This commit also adds a few entries to the .gitignore file.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "trailingComma": "es5",
+    "tabWidth": 2,
+    "semi": false,
```
Comment (spruce) [inline .prettierrc]: We use semicolons through-out the project. Why did you choose to remove all of them? Otherwise nice config.

Comment (maks-io) [reply .prettierrc]: I gotta admit I didn't spend too much time thinking about specific style choices - i actually should have created a draft PR instead.
Anyhow, I'll adjust this soon. Also the README part of course. Thanks for your feedback!

**Comment by maks-io** [comment]
Hi @spruce , sorry for the delay. I now adjusted the prettier config as you suggested and I also added two more dev dependencies: *husky* and *lint-staged*. The current project configuration will now make sure that prettier will always automatically run on all staged files, when a contributor wants to do a commit. Also I manually ran prettier on all already existing files so that we get one clean state.

**Comment by maks-io** [comment]
> This looks good. Do you wanna push any other changes into this PR?

No, that's it for now, thank you!

**Pull request merged**